rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -717,6 +717,16 @@ private predicate modImportNestedLookup(Module m, ModuleItemNode ancestor, Folde
   )
 }
 
+private predicate pathAttrImport(Folder f, Module m, string relativePath) {
+  exists(Meta meta |
+    f = m.getFile().getParentContainer() and
+    meta = getPathAttrMeta(m) and
+    relativePath = meta.getExpr().(LiteralExpr).getTextValue().regexpCapture("\"(.+)\"", 1)
+  )
+}
+
+private predicate append(Folder f, string relativePath) { pathAttrImport(f, _, relativePath) }
```

Comment:
This had me a bit confused at first as the predicate is called `append` but doesn't seem to append :)

If I understand correctly this is about limiting the things that are appended by `append` inside `Append` in order for it not to append everything in the world?

What about calling it `shouldAppend`/`limitAppend`/`appendOnly`/`shouldBeAppended` (or something along those lines) and then using the same name for the parameter to the `Append` module (what is now called `app`)?